planner/core/preprocess.go

```diff
@@ -52,6 +68,7 @@ type preprocessor struct {
 	tableAliasInJoin []map[string]interface{}
 
 	parentIsJoin bool
+	inTxRetry    bool
```

Comment:
1. s/inTxRetry/inTxnRetry/?
2. how about putting this variable together with `inPrepare` and adding comments about these two variables?